File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> num = new HashSet<Integer>();
        int ans = 0;
        
        for(int i : nums) {
            num.add(i);
        }
        
        for(int i : nums){
            if(!num.contains(i-1)){
                int count = 1;
                while(num.contains(i+count)){
                    count++;
                }
                ans = Math.max(ans, count);
            }
        }
        return ans;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num = set(nums)
        ans = 0
        for i in num:
            if i - 1 not in num:
                count = 1
                while (i+count) in num:
                    count += 1
                ans = max(ans, count)
        return ans